src/main/java/org/idpass/lite/IDPASSMap.java

```diff
@@ -140,31 +140,25 @@ public String get(String data) {
         String arr[];
 
         try {
-            switch (currentField) {
-                case "surName":
-                    arr = ret.split("\\s*(,|\\s)\\s*");
-                    if (ret.contains(",")) {
-                        ret = arr[0];
-                    } else {
-                        List<String> L = Arrays.asList(arr);
-                        ret = L.get(L.size() - 1);
-                    }
-                   break;
-
-                case "givenName":
-                    arr = ret.split("\\s*(,|\\s)\\s*");
-                    if (ret.contains(",")) {
-                        List<String> L = Arrays.asList(arr);
-                        ret = L.stream().skip(1).collect(Collectors.joining(" "));
-                    } else {
-                        ret = arr[0];
-                    }
-                    break;
-
-                case "dateOfBirth":
-                    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy/MM/d");
-                    LocalDate dob = LocalDate.parse(ret, formatter);
-                    break;
+            if (currentField.equals(surName.getValue())) {
+                arr = ret.split("\\s*(,|\\s)\\s*");
+                if (ret.contains(",")) {
+                    ret = arr[0];
+                } else {
+                    List<String> L = Arrays.asList(arr);
+                    ret = L.get(L.size() - 1);
+                }
+            } else if(currentField.equals(givenName.getValue())) {

```

Comment:
I don't think we should do this. This is too specific to a country. We should take the value as-is.